### DISCUSSION TITLE
Subscription not working with filters

### DISCUSSION THREAD

**Discussion by Talch87** [supabase/supabase#17595] [Questions] [answered]
Hi, 

my subscriptions works perfectly if I don't use filters. 
When I filter by variable it throws an error.
I'm on React/NextJs. I assume there's some error on how I wrote the variable filter, but the error message below seems to read the variable correctly (restaurant1 is the correct value)

Any help is highly appreciated!!
Thanks

**Without filter**
const subscription = supabase
    .channel('*')
    .on(
      'postgres_changes',
      {
        event: '*',
        schema: 'public',
        table: 'orders',
      },

**With filter**
```
const subscription = supabase
    .channel('*')
    .on(
      'postgres_changes',
      {
        event: '*',
        schema: 'public',
        table: 'orders',
        filter:`restaurant_id=eq.${restaurantId}`
      },
```

restaurantId is a string and the value is correct.
restaurant_id is the correct column name of table orders.

this is the error I get in Supabase logs:
Subscribing to PostgreSQL failed: {:error, "Subscription insert failed with error: ERROR P0001 (raise_exception) invalid column for filter restaurant_id. Check that tables are part of publication supabase_realtime and subscription params are correct: %{\"event\" => \"*\", \"filter\" => \"restaurant_id=eq.restaurant1\", \"schema\" => \"public\", \"table\" => \"orders\"}"}

**Comment by GaryAustin1** [comment]
Not sure if you have a typo but the filter is a string.
![image](https://github.com/supabase/supabase/assets/54564956/48818e6f-618e-4e0e-9d3b-cbdc16d9258a)

**Reply by GaryAustin1** [reply]
Did you attempt to make the filter a string with js template format?
```
filter:`restaurant_id=eq.${restaurantId}`
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

**Reply by Talch87** [reply]
yes but it's not working. It's written in that exact way in my original message. I've now edited the post formatting it as code.

**Reply by GaryAustin1** [reply]
Well that makes a big difference...
Can you show your table column definition for restaurant_id and the actual column name and type?

**Reply by GaryAustin1** [reply]
Also do you use Prisma, or custom JWTs?  And if you got back to no filter it still works currently?
The few cases I see of the error message seem to have broader issue than just the that particular column.

**Reply by Talch87** [reply]
this is my table definition.
Yes, If I remove the filter the subscription works perfectly.
I'm not using Prisma and no JWTs.

```
  public.orders (
    order_id text not null,
    restaurant_id text not null,
    table_id uuid not null,
    user_id text not null,
    order_datetime timestamp without time zone not null,
    order_type text null,
    order_status text not null default 'open'::text,
    order_amount real not null,
    order_vat real not null,
    table_number smallint not null,
    order_number smallint not null,
    status_updated timestamp without time zone null,
    notes text null,
    wait_time smallint null,
    stripe_payment_intent_id text null,
    constraint orders_pkey primary key (order_id),
    constraint orders_stripe_payment_intent_id_key unique (stripe_payment_intent_id),
    constraint orders_restaurant_id_fkey foreign key (restaurant_id) references restaurants (id),
    constraint orders_table_id_fkey foreign key (table_id) references tables (id)
  ) tablespace pg_default;
```

**Reply by GaryAustin1** [reply, marked as answer]
Looks fine.
Does a filter on another column work?
You may want to contact support or generate an issue in supabase/realtime.

**Reply by Talch87** [reply]
no, also other subscriptions have the same issue.
I've seen the demo on [...slug].tsx in the realtime library and it looks the same as what I did, so it's really a mistery why it's not working.

I'll keep trying.

Thanks a lot for your help and time!

**Comment by Talch87** [comment]
if the value is fixed then it's easy to put all as a string, but in my case I need the filter to match a variable restaurantId.
How can I pass that variable in the filter?

**Comment by Talch87** [comment]
can anybody help?
